core/os/semaphore.h

```diff
@@ -34,30 +34,32 @@
 #include "core/error_list.h"
 #include "core/typedefs.h"
 
+#if !(defined NO_THREADS)
```

Comment:
We'd usually write this `#if !defined(NO_THREADS)`.

Reply:
Fixed.